### ISSUE TITLE
Weird bahavior

### ISSUE THREAD

**Issue by forcel0** [doxygen/doxygen#10134]
**Describe the bug**
We compile documentation for a larger project including sources from C++, C# and others.  The C++ works great, the C# using mostly the same parameters is exhibiting a strange behavior.  It seem to not to pickup the object link in the C# files. It does link the method but stops linking after the first one. If I change the order of the lines of code the first line, different method, now links but subsequent do not. 

**Expected behavior**
I would expect all methods to link like the first.

**Screenshots**
Attached

**Version** 1.8.20

**Additional context**
I am setup to debug and step into the 1.8.20 Doxygen sources if there are some areas worth looking into.

![AxisObject](https://github.com/doxygen/doxygen/assets/42810251/763ee6f1-5f8c-461a-8591-127d94b084ac)
![AxisObject2](https://github.com/doxygen/doxygen/assets/42810251/3625e637-4e75-4151-a3cf-e6990b077f07)
![Cplus](https://github.com/doxygen/doxygen/assets/42810251/a2c441a5-1e9f-42aa-9915-8c3ae07f05cf)

**Comment by albert-github** [comment]
Doxygen version 1.8.20 is from August 24 2020 , the current version of doxygen is 1.9.7.

Images can be good for giving some further details on problems but are bad for showing code (e.g. one cannot do a cut / paste on it).

You wrote: 
> It seem to not to pickup the object link in the C# files. It does link the method but stops linking after the first one. If I change the order of the lines of code the first line, different method, now links but subsequent do not.?"

- where does this happen ? Source browser? Code snippets? Inline sources?

Furthermore:
- Can you check against the current doxygen version
- Can you please attach a, small, self contained example (source+configuration file in a, compressed, tar or zip file) that allows us to reproduce the problem? Please don't add external links as they might not be persistent.

**Comment by forcel0** [comment]
Fair enough.  I've upgraded to 1.9.7 and have most everything building but I am getting a behavior that was not present in 1.8.20.
We use the modules functionality to rearrange our navigation for classes and sample application under different namespaces. If we disable the Class in the LAYOUT_FILE it no longer generates the html for the class methods. It does generate the tree view but the end method html documentation is not generated. If Class is enabled then the documentation generates correctly but then Class shows up in the navigation as a standalone and also in the module.
[ClassHTML-NavIsIncorrect.zip](https://github.com/doxygen/doxygen/files/11782203/ClassHTML-NavIsIncorrect.zip)
[NoClassHTML-NavIsCorrect.zip](https://github.com/doxygen/doxygen/files/11782204/NoClassHTML-NavIsCorrect.zip)

**Comment by albert-github** [comment]
Looks like the issue in https://github.com/doxygen/doxygen/issues/10134#issuecomment-1596216869is different from the original issue (is the original issue solved?). It is always better to separate issues when they are not related (though the positive side is that there is an example illustration the issue).

Combination of groups and other entities are always difficult and especially when the names resemble each other. In test cases it is therefore always better to use unique names (e.g. in the group description use `G_Class` instead of `Class` etc. like in [ClassHTML-NavIsIncorrect_2.tar.gz](https://github.com/doxygen/doxygen/files/11798517/ClassHTML-NavIsIncorrect_2.tar.gz))

In the NoClass example it is in my opinion logical that there are no classes as they are explicitly excluded.

@doxygen what do you think?

**Comment by forcel0** [comment]
I would expect that since definitions for for class page, namespace page, file page, etc. all have  in the LAYOUT_FILE <classes visible="yes" title=""/> set the only place it should hide it from would be the <navidex> where it is set  <tab type="classes" visible="no" title="">. I would still expect it to generate the classes documentation.

We are able to to fix by setting <navidex> to <tab type="classes" visible="yes" title=""> so it generates the classes documentation and then hiding the navigation link with custom .css file under HTML_EXTRA_STYLESHEET

Point taken on unique class and other keyword names. Thanks.

[api_modules.zip](https://github.com/doxygen/doxygen/files/11800245/api_modules.zip)

**Comment by forcel0** [comment]
Inheritance works on our project now. We have one last issue with grouping and snippets to build up a document. I started new ticket as it is a separate problem. [#10148](https://github.com/doxygen/doxygen/issues/10148)

Thanks for changes in 1.9.8 to fix #10134
